eng/publish/publish.yml

```diff
@@ -0,0 +1,281 @@
+      # to push any package versions that were already uploaded to NuGet.
+      # For instance, imagine the latest DTFx packages on NuGet are: DTFx.Core@v1.1 and DTFx.AzureStorage@v1.0
+      # and that our official pipeline generated these packages: DTFx.Core@v1.1 and DTFx.AzureStorage@v1.1.
+      # In the end, we want NuGet to have both packages at version v1.1.
+      #
+      # In this case, if we try to publish both packages in the same ADO job, it will fail as soon as it detects
+      # we're trying to push DTFx.Core@v1.1 as that already exists on NuGet, and this could prevent
+      # DTFx.AzureStorage@v1.1 from being published in the first place.
+      # 
+      # Unfortunately, skipping duplicate package versions is only supported for publishing to ADO feeds, which explains
+      # why we did not have this issue when pushing to ADO.
+      # 
+      # So we split each package upload to a separate job, so they be uploaded independently. In our example, the
+      # DTFx.Core upload task would fail (as expected, it was already uploaded), 
+      # but the DTFx.AzureStorage upload would succeed.
+
+      # Implementor's note:
+      # What follows could use some refactoring as there's a lot of repetition, perhaps an ADO template could hide the repeated details.
+      # Pay close attention to the `packagesToPush` property of each job. We need to make sure the pattern specified there uniquely identifies
+      # the package we want to push, and this is difficult when the full name of one package is the prefix of another.
+      # For example: `Microsoft.DurableTask.Client` and `Microsoft.DurableTask.Client.Grpc`.
+      # The trick is to exclude the `Microsoft.DurableTask.Client.Grpc` in the `packagesToPush` property for `Microsoft.DurableTask.Client`
+
+      # NuGet release (Microsoft.DurableTask.Abstractions)
+      - job: nugetRelease_Microsoft_DurableTask_Abstractions
+        displayName: NuGet Release (Microsoft.DurableTask.Abstractions)
+        dependsOn: nugetApproval
+        condition: succeeded('nugetApproval') # nuget packages need to be on ADO first
+        templateContext:
+          type: releaseJob
+          isProduction: true
+          inputs:
+          - input: pipelineArtifact
+            pipeline: officialPipeline  # Pipeline reference as defined in the resources section
+            artifactName: drop
+            targetPath: $(System.DefaultWorkingDirectory)/drop
+        steps:
+        - task: 1ES.PublishNuget@1
+          displayName: 'NuGet push (Microsoft.DurableTask.Abstractions)'
+          inputs:
+            command: push
+            nuGetFeedType: external
+            publishFeedCredentials: 'DurableTask org NuGet API Key'
+            packagesToPush: '$(System.DefaultWorkingDirectory)/drop/Microsoft.DurableTask.Abstractions.*.nupkg;!$(System.DefaultWorkingDirectory)/**/*.symbols.nupkg' # Despite this being a custom command, we need to keep this for 1ES validation

```

Comment:
`!$(System.DefaultWorkingDirectory)/**/*.symbols.nupkg` needs to be included?` Weird - I don't believe we produce those. Instead we produce `.snupkg` (which I believe will automatically be included when a nupkg is pushed)

Reply:
The `/**/*.symbols.nupkg` bit appears here because the default value for `packagesToPush` includes it, and I was just modifying that default template. You're right, we should be able to remove it; let me know into that